object/ldap_conn.go

```diff
@@ -346,31 +346,52 @@ func SyncLdapUsers(owner string, syncUsers []LdapUser, ldapId string) (existUser
 	}
 	tag := strings.Join(ou, ".")
 
-	for _, syncUser := range syncUsers {
-		existUuids, err := GetExistUuids(owner, uuids)
-		if err != nil {
-			return nil, nil, err
-		}
+	ldapUuids, err := GetExistUuids(owner, uuids)
+	if err != nil {
+		return nil, err
+	}
 
-		found := false
-		if len(existUuids) > 0 {
-			for _, existUuid := range existUuids {
-				if syncUser.Uuid == existUuid {
-					existUsers = append(existUsers, syncUser)
-					found = true
+	for _, syncUser := range syncUsers {
+		foundLdapUuid := ""
+		if len(ldapUuids) > 0 {
+			for _, ldapUuid := range ldapUuids {
+				if syncUser.Uuid == ldapUuid {
+					foundLdapUuid = ldapUuid
 				}
 			}
 		}
 
 		name, err := syncUser.buildLdapUserName()
 		if err != nil {
-			return nil, nil, err
+			return nil, err
 		}
 
-		if !found {
+		var affected bool
+		if foundLdapUuid != "" {
+			existUser, err := GetUserByLdapUuid(owner, foundLdapUuid)
+			if err != nil {
+				return nil, err
+			}
+
+			existUser.Owner = owner
+			existUser.Name = name
+			existUser.DisplayName = syncUser.buildLdapDisplayName()
+			existUser.Email = syncUser.Email
+			existUser.Phone = syncUser.Mobile
+			existUser.Address = []string{syncUser.Address}
+			existUser.Affiliation = affiliation
+			existUser.Tag = tag
+			existUser.Ldap = syncUser.Uuid
```

Comment:
Why we should update owner, name,ldap fields? I don't sure it's good idea to update name of existing user.

Reply:
if mapping field was changed, shouldn't we change user properties with new value of mapping field? I think this is truly synchronization

Reply:
name is calculated field that could be different from ldap

Reply:
fixed

Reply:
But anyway, I think we should have discussions about sync logic at all. I don't sure we should sync attributes/roles always if ldap sync enabled.